iroh-net/src/relay/server.rs

```diff
@@ -510,6 +516,44 @@ impl AsyncWrite for MaybeTlsStream {
     }
 }
 
+pub struct ClientCounter {
+    clients: HashMap<PublicKey, usize>,
+    last_clear_date: Date,
+}
+
+impl ClientCounter {
+    /// Creates a new client counter.
+    pub fn new() -> Self {
+        Self {
+            clients: HashMap::new(),
+            last_clear_date: OffsetDateTime::now_utc().date(),
+        }
+    }
+
+    fn check_and_clear(&mut self) {
+        let today = OffsetDateTime::now_utc().date();
+        if today != self.last_clear_date {
+            self.clients.clear();
+            self.last_clear_date = today;
+        }
+    }
+
+    /// Updates the client counter.
+    pub fn update(&mut self, client: PublicKey) -> u64 {
+        self.check_and_clear();
+        let mut new_conn = true;
+        if self.clients.contains_key(&client) {
+            new_conn = false;
+        }
+        let counter = self.clients.entry(client).or_insert(0);
+        *counter += 1;
+        match new_conn {
+            true => 1,
+            false => 0,
+        }
```

Comment:
```rust
new_conn as u64
```

Reply:
I generally dislike that as I never feel I can trust that implicit conversion from bool to int in regards to compiler/platforms.